### ISSUE TITLE
Planning ideas etc 

### ISSUE THREAD

**Issue by banillie** [yulqen/dbasik#10]
Hi @yulqen I just updated my local version of dbasik as I wanted to have a test of it again. I see you've not pushed the latest code into the master branch since March. Is there a newer version that I can pull down easily. Not sure if you were working on this Django version in your latest code. Anyway I wanted to pull down the latest version to have a play around with.

I've done some thinking off the back of what we discussed last week in terms of next steps. All the ideas below are geared to the import (rather than export) side of things and there is a deadline to work to if we want to have things in place by then - to set ourselves a bit of a challenge - which is Wednesday 6 July! So the ideas are:

- Create api endpoint's which supply data as per datamaps api `project_data_from_master` but as json not a python dictionary. These end points could be at localhost to start with. This could be in dbasik or datamaps. 
- analysis_engine to work as an api client to get data and produce current graphical outputs.  Essentially removing use of `project_data_from_master`. 

These seem to me to be the obvious first steps and might be achievable by 6 July? 

Other things we could do afterwards are:
- Build graphical outputs into dbasik. 
- Build a new api client to product outputs. 
- Both the above could use existing code in analysis_engine or we could update /create new code. 

One further thing that doesn't really relate to the above, but is something worth considering is having the option in the datamaps/dbasik interface to do a bulk upload of master data (i.e. not just individual returns). On the basis that it may be helpful in some instances, where data is held on a single excel spreadsheet (which has probably been created by datamaps!), to upload all the data. 

Let me know what you think. 

Cheers.

**Comment by yulqen** [comment]
@banillie - thanks, and all good ideas which I support. I welcome the deadline too.

I've merged recent development in to master now so you can get up to date. Let me know if you have any problems firing it up. You should literally be able to do:

- `docker-compose up`. Let it spin up. You should be able to get the site on localhost, but you most likely won't be able to log in.
- `make createsuperuer` - to create an admin account
- `make create-financial-quarters` - to generate some test FYs in the system.

Let me know how it goes.

**Comment by banillie** [comment]
@yulqen thanks. 

That's all working. I had to do the `make clean-rebuild-and-run` cmd initially before I could get the `docker-compose up` cmd to work, but I guess that makes sense as the initial command triggered a whole load of installations. 

The local version that loaded kept all the previous data in the db e.g. my username and password, and some returns I had uploaded, and financial quarters. 

I did at one point do `pip install -r requirements.txt` which got a number of errors. But presumably this isn't a problem as the environment is built in docker rather than the venv? 

Unfortunately I need to leave at 4 tomorrow. Is there any chance we could bring the coding session forward one hour to start at 2.30?

**Comment by yulqen** [comment]
In terms of syncing up with `analysis-engine`, we have two choices:

1. We tailor the dbasik API to fit `analysis-engine`, or
2. We tailor `analysis-engine` to fit dbasik.

I think the best approach is going to be to do a standard REST-type API from dbasik that you can tailor `analysis-engine` to use. dbasik will mirror the way you do things in `analysis-engine` to an extent anyway - dbasik has a project register, which contains all the top-level project information a la your spreadsheet, plus the return data. There will just be a bit of shuffling around to do.

So I think if I get the basic `returns` and `register` API stuff going, that exposes everything, we can finesse it as require, but that will give us the opportunity to work on `analysis-engine` to bring it into shape with the new source of data. We can work on that together or I can help. Either way - plenty to do.

**Comment by banillie** [comment]
Sounds great. I agree it make sense for Dbasik to supply api to analysis_engine as a first step. In effect dbasik will be replacing datamaps (the compilation of data) and analysis_engine (provision of data in required format). So all analysis_engine will do is provide the outputs. I think that will be a really useful first step and we can take it from there afterwards. 
Cheers!

**Comment by banillie** [comment]
@yulqen would it be possible to list each quarters the project returns api data in the following format. 

The URL would be something like `[localhost:8000 or website_name]/projectreturns/[quarter or month]`

[
{ID:
project_name:


##Ignore for now. will finish later!

**Comment by yulqen** [comment]
Yeah, no problem. Just let me know when you've completed the representation of the data structure you need out of it.
